pkgs/by-name/rb/rbw/package.nix

```
@@ -54,6 +51,23 @@ rustPlatform.buildRustPackage rec {
       --fish <($out/bin/rbw gen-completions fish) \
       --zsh <($out/bin/rbw gen-completions zsh)
   ''
+  + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    for gen in nushell powershell elvish fig; do
+      case "$gen" in
+        nushell) dir=share/nushell/completions; file=rbw.nu ;;
+        powershell) dir=share/powershell/completions; file=rbw.ps1 ;;
+        elvish)     dir=share/elvish/completions;     file=rbw.elv ;;
+        fig)        dir=share/fig/completions;        file=rbw.ts ;;
+      esac
+
+      echo "rbw: generating $gen completion..." >&2
+      payload="$($out/bin/rbw gen-completions "$gen" 2>/dev/null || true)"
+
+      install -d "$out/$dir"
+      printf '%s\n' "$payload" > "$out/$dir/$file"
+      echo "rbw: wrote $gen completion to $dir/$file" >&2
+    done
```

```suggestion
    for shell in nushell powershell elvish fig; do
      case "$shell" in
        nushell)    dir=share/nushell/completions;    file=rbw.nu ;;
        powershell) dir=share/powershell/completions; file=rbw.ps1 ;;
        elvish)     dir=share/elvish/completions;     file=rbw.elv ;;
        fig)        dir=share/fig/completions;        file=rbw.ts ;;
      esac

      echo "rbw: generating $shell completion..." >&2

      install -d "$out/$dir"
      echo "$($out/bin/rbw gen-completions "$shell" 2>/dev/null || true)" > "$out/$dir/$file"
    done
```

it would be nicer to use the standard pattern like https://github.com/NixOS/nixpkgs/blob/a421ac6595024edcfbb1ef950a3712b89161c359/pkgs/applications/networking/cluster/helmfile/default.nix#L43-L46 but for that we would need to add evlish and fig support there.